gif-screencast.el

```diff
@@ -121,13 +121,83 @@ various programs run here."
   :init-value nil
   :keymap gif-screencast-mode-map)
 
+(defvar gif-screencast--done nil
+  "Flag if capturing already done.")
+
+(defvar gif-screencast--finishing nil
+  "Flag if compilation in progress.")
+
+(defvar gif-screencast--counter 0
+  "Number of running screenshots.")
+
+(defun gif-screencast-capture-sentinel (_proc _status)
+  "Sentinel for screen capturing."
+  (setq gif-screencast--counter (- gif-screencast--counter 1))
+  (gif-screencast--finish))
+
+(defun gif-screencast-done ()
+  "Return t if screen capturing already done."
+  (and gif-screencast--done (= gif-screencast--counter 0)))
+
+(defun gif-screencast--finish ()
+  "Finish screen capturing."
+  (if (gif-screencast-done)
+      (if gif-screencast--finishing
```

Comment:
What' the purpose of `gif-screencast--finishing`?

Reply:
For prevent double compilation

Reply:
Can you detail how this can possibly happen?

Reply:
There is two places where we call `gif-screencast--finish`. Sometimes `(gif-screencast-done)` can be true for all of it. So we can start compilation twice.

Reply:
Correct me if I'm wrong:

Whenever we are in `gif-screencast--finish`, the execution cannot be interrupted.
- When sentinels call it and the screencast wasn't stopped by the user, `gif-sceencast-done` returns nil.
- When sentinels call it and the screencast was stopped by the user, `gif-sceencast-done` returns non-nil and the function runs.
- When the user stops the execution and at least one capture is still processing, some sentinels still haven't decreased the counter so `gif-screencast-done` is nil.
- When the user stops the execution and all captures have been processed, all the sentinels have decreased the counter down to 0 so `gif-screencast-done` is non-nil.

Only possible ways to go pass `gif-screencast-done` and those are mutually exclusive unless I'm mistaken.

Reply:
Looks like you are right.